plugins/modules/foreman_location.py

```diff
@@ -131,29 +118,49 @@ def main():
         supports_check_mode=True,
     )
 
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    name_or_title = entity_dict.pop('name')
-    parent = entity_dict.pop('parent', None)
+    # Hack for broken foreman api
+    _location_create = next(x for x in module.foremanapi.apidoc['docs']['resources']['locations']['methods'] if x['name'] == 'create')
+    _location_create_params_location = next(x for x in _location_create['params'] if x['name'] == 'location')
+    _location_create_params_location['params'].append({
+        u'validations': [],
+        u'name': u'organization_ids',
+        u'show': True,
+        u'description': u'\n<p>Organization IDs</p>\n',
+        u'required': False,
+        u'allow_nil': True,
+        u'allow_blank': False,
+        u'full_name': u'location[organization_ids]',
+        u'expected_type': u'array',
+        u'metadata': None,
+        u'validator': u'',
+    })
 
     # Get short name and parent from provided name
-    parent_from_title, name = split_fqn(name_or_title)
-
+    name, parent = split_fqn(entity_dict['name'])
     entity_dict['name'] = name
+
+    if 'parent' in entity_dict:
+        if parent:
+            module.fail_json(msg="Please specify the parent either separately, or as part of the title.")
+        parent = entity_dict['parent']
     if parent:
-        entity_dict['parent'] = find_location(module, parent)
-    elif parent_from_title:
-        entity_dict['parent'] = find_location(module, parent_from_title)
+        entity_dict['parent'] = module.find_resource('locations', search='title="{}"'.format(parent), thin=True, failsafe=module.desired_absent)
+
+        if module.desired_absent and entity_dict['parent'] is None:
+            # Parent location does not exist so just exit here
+            module.exit_json(changed=False)
 
-    if 'organizations' in entity_dict:
-        entity_dict['organizations'] = find_organizations(module, entity_dict['organizations'])
+    if not module.desired_absent:
+        if 'organizations' in entity_dict:
+            entity_dict['organizations'] = module.find_resources('organizations', entity_dict['organizations'], thin=True)
 
-    entity = find_location(module, title=build_fqn(name_or_title, parent), failsafe=True)
-    entity_dict = sanitize_entity_dict(entity_dict, name_map)
+    entity = module.find_resource('locations', search='title="{}"'.format(build_fqn(name, parent)), failsafe=True)
```

Comment:
I think I'd prefer if `search='title="{}"'.format(build_fqn(name, parent))` would be an own line, and then you'd call with `search=search` as format strings inside of args are hard to read.